examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -91,7 +91,9 @@ void RvcDevice::HandleRvcCleanChangeToMode(uint8_t newMode, ModeBase::Commands::
 
 void RvcDevice::HandleOpStatePauseCallback(Clusters::OperationalState::GenericOperationalError & err)
 {
-    auto error = mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kPaused));
+    // This method is only called if the device is in a Pause-compatible state, i.e. `Running` or `SeekingCharger`.
+    mRusumeState = mOperationalStateInstance.GetCurrentOperationalState();
```

Comment:
From the state machine diagram, please note that there are states that we can resume from but we cannot pause. This meas that we cannot holistically update `mRusumeState` in this method if it is to be use as you are suggesting.

To do what you are asking for, we would need to keep updating `mResumeState` whenever the combination of state and run mode would allow for the resume command to be accepted. This would scatter this logic in a number of places.

To my mind it makes it simpler to check the situation that we are in when the resume command is received.